Sources/Network/BoxNetworkTask.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  BoxNetworkTask.swift
+//  BoxSDK-iOS
+//
+//  Created by Sujay Garlanka on 3/21/20.
+//  Copyright © 2020 box. All rights reserved.
+//
+
+import Foundation
+
+/// A Box network task request.
+public class BoxNetworkTask {
+
+    var task: URLSessionTask?
```

Comment:
If we only ever use this for cancellation, and we don't expect `task` and `nestedTask` to be alive at the same time (I'm not sure this is true), and it's not part of the public interface, have we considered that we don't need the three task types, and we could get away with:

```swift
// This doesn't need to be public, and probably shouldn't be, as it shares a name with
// a protocol in Combine.
protocol Cancellable {
    func cancel()
}

public class BoxNetworkTask {
    private var task: Cancellable?
    // …
}
```
This would be forwards compatible with deciding to add subclasses in the future. Maybe we already know we'll want them?

Reply:
We are thinking that BoxDownloadTask can have pause and resuming for downloads. Not sure what we could add to BoxUploadTasks yet, but thought it was better to have a separate class for that in case.